x/oracle/types/keys_test.go

```diff
@@ -0,0 +1,108 @@
+	testCases := []struct {
+		denom       string
+		expectedKey []byte
+	}{
+		{
+			denom:       app.BondDenom,
+			expectedKey: []byte{0x1, 0x75, 0x75, 0x6d, 0x65, 0x65, 0x0},
+		},
+		{
+			denom:       denomLunaIBC,
+			expectedKey: []byte{0x1, 0x69, 0x62, 0x63, 0x2f, 0x30, 0x45, 0x46, 0x31, 0x35, 0x44, 0x46, 0x32, 0x46, 0x30, 0x32, 0x34, 0x38, 0x30, 0x41, 0x44, 0x45, 0x30, 0x42, 0x42, 0x36, 0x45, 0x38, 0x35, 0x44, 0x39, 0x45, 0x42, 0x42, 0x35, 0x44, 0x41, 0x45, 0x41, 0x32, 0x38, 0x33, 0x36, 0x44, 0x33, 0x38, 0x36, 0x30, 0x45, 0x39, 0x46, 0x39, 0x37, 0x46, 0x39, 0x41, 0x41, 0x44, 0x45, 0x34, 0x46, 0x35, 0x37, 0x41, 0x33, 0x31, 0x41, 0x41, 0x30, 0x0},
+		},
+		{
+			denom:       denomAtomIBC,
+			expectedKey: []byte{0x1, 0x69, 0x62, 0x63, 0x2f, 0x32, 0x37, 0x33, 0x39, 0x34, 0x46, 0x42, 0x30, 0x39, 0x32, 0x44, 0x32, 0x45, 0x43, 0x43, 0x44, 0x35, 0x36, 0x31, 0x32, 0x33, 0x43, 0x37, 0x34, 0x46, 0x33, 0x36, 0x45, 0x34, 0x43, 0x31, 0x46, 0x39, 0x32, 0x36, 0x30, 0x30, 0x31, 0x43, 0x45, 0x41, 0x44, 0x41, 0x39, 0x43, 0x41, 0x39, 0x37, 0x45, 0x41, 0x36, 0x32, 0x32, 0x42, 0x32, 0x35, 0x46, 0x34, 0x31, 0x45, 0x35, 0x45, 0x42, 0x32, 0x0},
+		},
+	}
+
+	for i, testCase := range testCases {
+		actualKey := types.GetExchangeRateKey(testCase.denom)
+		require.Equalf(t, testCase.expectedKey, actualKey, "test %d - expected key: %s should be the same as actual key: %s", i, testCase.expectedKey, actualKey)
+	}
+}
+
+func TestGetFeederDelegationKey(t *testing.T) {
+	testCases := []struct {
+		val         sdk.ValAddress
+		expectedKey []byte
+	}{
+		{
+			val:         []byte("addr________________"),
+			expectedKey: []byte{0x2, 0x14, 0x61, 0x64, 0x64, 0x72, 0x5f, 0x5f, 0x5f, 0x5f, 0x5f, 0x5f, 0x5f, 0x5f, 0x5f, 0x5f, 0x5f, 0x5f, 0x5f, 0x5f, 0x5f, 0x5f},
+		},
+	}
+
+	for i, testCase := range testCases {
+		actualKey := types.GetFeederDelegationKey(testCase.val)
+		require.Equalf(t, testCase.expectedKey, actualKey, "test %d - expected key: %s should be the same as actual key: %s", i, testCase.expectedKey, actualKey)
+	}
+}
+
+func TestGetMissCounterKey(t *testing.T) {
+	testCases := []struct {
+		val         sdk.ValAddress
+		expectedKey []byte
+	}{
+		{
+			val:         []byte("addr________________"),
+			expectedKey: []byte{0x3, 0x14, 0x61, 0x64, 0x64, 0x72, 0x5f, 0x5f, 0x5f, 0x5f, 0x5f, 0x5f, 0x5f, 0x5f, 0x5f, 0x5f, 0x5f, 0x5f, 0x5f, 0x5f, 0x5f, 0x5f},
+		},
+	}
+
+	for i, testCase := range testCases {
+		actualKey := types.GetMissCounterKey(testCase.val)
+		require.Equalf(t, testCase.expectedKey, actualKey, "test %d - expected key: %s should be the same as actual key: %s", i, testCase.expectedKey, actualKey)
+	}
+}
+
+func TestGetAggregateExchangeRatePrevoteKey(t *testing.T) {
+	testCases := []struct {
+		val         sdk.ValAddress
+		expectedKey []byte
+	}{
+		{
+			val:         []byte("addr________________"),
+			expectedKey: []byte{0x4, 0x14, 0x61, 0x64, 0x64, 0x72, 0x5f, 0x5f, 0x5f, 0x5f, 0x5f, 0x5f, 0x5f, 0x5f, 0x5f, 0x5f, 0x5f, 0x5f, 0x5f, 0x5f, 0x5f, 0x5f},
```

Comment:
Could add a comment as to the origin of these bytes - that would also clarify what the meaning would be if the test ever failed.

Reply:
For example:
```
// keyPrefixAggregateExchangeRatePrevote | lengthPrefixed(addr)
```